src/wh_client_cryptocb.c

```diff
@@ -1009,6 +1012,155 @@ static int _xferSha256BlockAndUpdateDigest(whClientContext* ctx,
     return ret;
 }
 
+#ifdef WOLFHSM_CFG_DMA
+
+static int _handleSha256Dma(int devId, wc_CryptoInfo* info, void* inCtx,
+                            whPacket* packet)
+{
+    int              ret    = 0;
+    whClientContext* ctx    = inCtx;
+    wc_Sha256*       sha256 = info->hash.sha256;
+    uint16_t         respSz = 0;
+    uint16_t         group  = WH_MESSAGE_GROUP_CRYPTO_DMA;
+
+#if WH_DMA_IS_32BIT
+    wh_Packet_hash_sha256_Dma32_req* req   = &packet->hashSha256Dma32Req;
+    wh_Packet_hash_sha256_Dma32_res* resp  = &packet->hashSha256Dma32Res;
+#else
+    wh_Packet_hash_sha256_Dma64_req* req   = &packet->hashSha256Dma64Req;
+    wh_Packet_hash_sha256_Dma64_res* resp  = &packet->hashSha256Dma64Res;
+#endif
+
+    /* Caller invoked SHA Update:
```

Comment:
Yep, since client calls wc_Sha256Init() on the context already, and state is held by client, it already has been initialized. CryptoCb doesn't hook the init anyway. Agreed that if state was held by server it would need to be initialized remotely as well, but here the client has already initialized.